python/semantic_kernel/kernel.py

```diff
@@ -235,47 +239,82 @@ def _create_semantic_function(
         # TODO: allow to postpone this (use lazy init)
         # allow to create semantic functions without
         # a default backend
-        backend = self._config.get_completion_backend(
-            function_config.prompt_template_config.default_backends[0]
-            if len(function_config.prompt_template_config.default_backends) > 0
-            else None
-        )
-
-        function.set_ai_configuration(
-            CompleteRequestSettings.from_completion_config(
-                function_config.prompt_template_config.completion
+        if function_config.has_chat_prompt:
+            backend = self._config.get_chat_backend(
+                function_config.prompt_template_config.default_backends[0]
+                if len(function_config.prompt_template_config.default_backends) > 0
+                else None
             )
-        )
 
-        if backend.backend_type == BackendType.AzureOpenAI:
-            Verify.not_null(
-                backend.azure_open_ai, "Azure OpenAI configuration is missing"
-            )
-            function.set_ai_backend(
-                lambda: AzureTextCompletion(
-                    backend.azure_open_ai.deployment_name,  # type: ignore
-                    backend.azure_open_ai.endpoint,  # type: ignore
-                    backend.azure_open_ai.api_key,  # type: ignore
-                    backend.azure_open_ai.api_version,  # type: ignore
-                    self._log,
+            function.set_chat_configuration(
+                ChatRequestSettings.from_completion_config(
+                    function_config.prompt_template_config.completion
                 )
             )
-        elif backend.backend_type == BackendType.OpenAI:
-            Verify.not_null(backend.open_ai, "OpenAI configuration is missing")
-            function.set_ai_backend(
-                lambda: OpenAITextCompletion(
-                    backend.open_ai.model_id,  # type: ignore
-                    backend.open_ai.api_key,  # type: ignore
-                    backend.open_ai.org_id,  # type: ignore
-                    self._log,
+
+            if backend.backend_type == BackendType.AzureOpenAI:
+                # TODO: azure impl
+                pass
```

Comment:
It's better to raise NotImplementedError in these cases so that it's obvious that the code needs to be implemented.